apps/cowswap-frontend/src/modules/ordersTable/pure/ReceiptModal/FeeField.tsx

```diff
@@ -7,23 +7,42 @@ import * as styledEl from './styled'
 
 export type Props = { order: ParsedOrder }
 
+// TODO: get rid of this once https://github.com/cowprotocol/services/pull/3184 is complete
+const HAS_BACKEND_MIGRATED = false
+
+function getFeeToken(order: ParsedOrder) {
+  if (!HAS_BACKEND_MIGRATED) {
+    return order.inputToken
+  }
+
+  const { inputToken, outputToken } = order
+  const { executedFeeToken } = order.executionData
+
+  if (inputToken?.address.toLowerCase() === executedFeeToken?.toLowerCase()) {
```

Comment:
if you want to be more fancy... 

```ts
const executedFeeTokenLower = executedFeeToken?.toLowerCase()
[inputToken, outputToken].find(token => token?.address.toLowerCase() === executedFeeTokenLower) 
```

Reply:
I dislike unnecessary arrays and iterating over arrays.

Also, you suggest to default to inputToken when executedFeeToken is not set?
Shouldn't happen, but makes sense to be backwards compatible I guess.

Will see what I can do.

Reply:
See 4e489f2de30e and d822798b5bd3

Reply:
Looks great, and simpler imo (as it has less repetition), but maybe its just a matter of taste. It was just nitpicking. Thanks for addressing though!